### PR TITLE
Upgrade to cabal-3.8.1.0

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -183,3 +183,9 @@ source-repository-package
   subdir:
     cardano-prelude
     cardano-prelude-test
+
+source-repository-package
+  type: git
+  location: https://github.com/astanin/moo
+  tag: dbda5e76ac3b4c72c805ec0cdb9bcdff7bb6247d
+  --sha256: 1mdj218hgh7s5a6b9k14vg9i06zxah0wa42ycdgh245izs8nfv0x
```
Comment (LudvikGalois) [inline cabal.project:191]: Is this needed (when using CHaP)?

Comment (michaelpj) [reply cabal.project:191]: It is not, there is a patched version in CHaP. Also please don't add more without a _clear_ justification!

Comment (newhoggy) [reply cabal.project:191]: When I tried last it the build failed on `moo`.  I'll create another PR without that commit to see if it happens again.

Comment (newhoggy) [reply cabal.project:191]: Actually I found the failed build here: https://github.com/input-output-hk/cardano-node/actions/runs/3306791861/jobs/5457943706

Comment (newhoggy) [reply cabal.project]: I updated the CHaP index state and it works now.

**Review by michaelpj — APPROVED** [review]

**Review by erikd — APPROVED** [review]

**Review by LudvikGalois — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**